apps/api/src/app/step-schemas/usecases/get-step-schema/get-step-schema.usecase.ts

```diff
@@ -1,39 +1,149 @@
+  (command as GetExistingStepSchemaCommand).workflowId !== undefined;
+
+export const buildControlsSchema = (stepType: StepType): Schema => {
+  return {
+    ...mapStepTypeToOutput[stepType],
+    description: 'Output of the step, including any controls defined in the Bridge App',
+  };
 };
 
-export const StepTypeToOutputMap = {
-  [ChannelStepEnum.SMS]: channelStepSchemas[ChannelStepEnum.SMS].output,
-  [ChannelStepEnum.EMAIL]: channelStepSchemas[ChannelStepEnum.EMAIL].output,
-  [ChannelStepEnum.PUSH]: channelStepSchemas[ChannelStepEnum.PUSH].output,
-  [ChannelStepEnum.CHAT]: channelStepSchemas[ChannelStepEnum.CHAT].output,
-  [ChannelStepEnum.IN_APP]: channelStepSchemas[ChannelStepEnum.IN_APP].output,
-  [ActionStepEnum.DELAY]: actionStepSchemas[ActionStepEnum.DELAY].output,
-  [ActionStepEnum.DIGEST]: actionStepSchemas[ActionStepEnum.DIGEST].output,
-  [ActionStepEnum.CUSTOM]: {
+const buildSubscriberSchema = () =>
+  ({
+    type: 'object',
+    description: 'Schema representing the subscriber entity',
+    properties: {
+      firstName: { type: 'string', description: "Subscriber's first name" },
+      lastName: { type: 'string', description: "Subscriber's last name" },
+      email: { type: 'string', description: "Subscriber's email address" },
+      phone: { type: 'string', description: "Subscriber's phone number (optional)" },
+      avatar: { type: 'string', description: "URL to the subscriber's avatar image (optional)" },
+      locale: { type: 'string', description: 'Locale for the subscriber (optional)' },
+      subscriberId: { type: 'string', description: 'Unique identifier for the subscriber' },
+      isOnline: { type: 'boolean', description: 'Indicates if the subscriber is online (optional)' },
+      lastOnlineAt: {
+        type: 'string',
+        format: 'date-time',
+        description: 'The last time the subscriber was online (optional)',
+      },
+    },
+    required: ['firstName', 'lastName', 'email', 'subscriberId'],
+    additionalProperties: false,
+  }) as const satisfies Schema;
+
+const buildPayloadSchema = () =>
+  ({
     type: 'object',
     properties: {},
     required: [],
     additionalProperties: false,
-  } as const satisfies Schema,
+    description:
+      'Payload Schema - For Developers. Passed during the novu.trigger method, and controlled by the developer.',
+  }) as const satisfies Schema;
+
+const buildVariablesSchema = (previousSteps?: NotificationStepEntity[]): Schema => {
+  return {
+    type: 'object',
+    description:
+      // eslint-disable-next-line max-len
+      'Variables that can be used with Liquid JS Template syntax. Includes subscriber attributes, payload variables, and supports liquid filters for formatting.',
+    properties: {
+      subscriber: buildSubscriberSchema(),
+      payload: buildPayloadSchema(),
+      steps: buildPreviousStepsSchema(previousSteps),
+    },
+    required: ['subscriber', 'payload'],
```

Comment:
What’s the purpose of the required field in the schema here? What are we trying to convey to the front end? Since it's a full object, not just a single variable, I’m curious about its necessity.